openssl-dynamic/src/main/java/org/apache/tomcat/jni/SSL.java

```diff
@@ -531,8 +531,16 @@ public static native long newBIO(long pool, BIOCallback callback)
      * the provided SSL instance, you must call freeBIO on the returned network BIO.
      *
      * @param ssl the SSL instance (SSL *)
+     * @param maxBioSize The maximum size of the BIO.
      * @return pointer to the Network BIO (BIO *)
      */
+    public static native long makeNetworkBIOWithSize(long ssl, int maxBioSize);
+
+    /**
+     * Creates a BIO with the default max BIO size.
+     *
+     * @see #makeNetworkBIOWithSize(long, int)
+     */
     public static native long makeNetworkBIO(long ssl);
```

Comment:
@siyengar can you just rename this method to `makeNetworkBIO0` and have it take `maxBioSize `. Then add static public methods to preserve old API:

Something like this:

```java
public static long makeNetworkBIO(long ssl) {
    return makeNetworkBIO0(ssl, maxBioSize);
}

public static long makeNetworkBIO(long ssl, int maxBioSize) {
    return makeNetworkBIO0(ssl, maxBioSize);
}

private static native long makeNetworkBIO0(long ssl, int maxBioSize);
```

This way we can keep the changes / c code minimal. WDYT ?